### ISSUE TITLE
Klondike options

### ISSUE THREAD

**Issue by jvanorsouw** [TobiasBielefeld/Simple-Solitaire#131]
Hi,

I am playing klondike a lot :-) and noticed several things which each may be a separate request and of which they might apply to other solitaire games.

- Empty spots are only fillable by a king, something which in some games leads to an unwinnable situation. And a rule which was allowed when I learned it years ago.
- Total hiding of scoring, because I play the game for fun so even hinting at statistics activates my wanting to win mode, instead of simple relaxation
- Already seen a similar issue: implementarion of a heuristic/algorithm to only generate winnable games. To shift impact this list could be generated during compilation as a hash table. New game would compare the generated game with the lookup table and try again if it is not invthat that winnable game list.

These items can be parameterized to not break other peoples experience of course

Thanks

**Comment by TobiasBielefeld** [comment]
hello, 
to your first point: Sure, I can add this as an settings option, will be in my next release. But I guess this makes the games way too easy to win.

to your second point: hiding the score is already possible in the appearance settings, but hiding the score inside the statistics is pretty strange. Well I can still add an option there to hide the win statistics panel. I hope this will be enough

to your third point: the feature for that is nearly finished and can already be tested with my testing branch or with this apk signed with my Google Play key: https://github.com/TobiasBielefeld/Simple-Solitaire/blob/testing/app/release/app-release.apk
You can install that apk file if you don't want to wait for the new release. I need more time for testing and I have currently a lot to do with work/uni, so nearly no free time for the app development :)

Currently the feature works like this:
* There is a new option in the general settings  called "Ensure movability". enable it
* In the expert settings is the entry "Ensure movability: Amount of moves". Click on it and enter a new value for Klondike. 500 is a good value
* then before every start, a new deal is tested if 500 moves are possible or the game is winnable within 500 moves. It checks deals until you cancel the process or it finds a suitable deal.
* 500 is a big number, so you will only get winnable games.

Pre calculating winnable games is not a thing I would like to have. Mostly because it would increase the app size (think of all possible rule combinations for all games, each would need an own set of winnable games)

**Comment by jvanorsouw** [comment]
Thanks for your quick response!

I downloaded the test version and noticed no delay on my Galaxy S5 using larger move values to try. This makes a precalculation indeed quite unnecessary.

Did you also improve the automove using this same approach, because I noticed a new option in the dev menu. I tried with 3 Pyramid games and the automove didnt succeed on those precalculated games. Klondike did however.

For my first request I just came up with the _nice_ _to_ _have_ extra functionality where you can just select the highest allowed card on the empty spaces e.g. limit it to picture cards only.

Hiding the statistics button would suffice for me in addition the already available option. Nothing fancy.

Again great that you are willing to spend time, good luck with your studies and very much thanks! Now back to trying to win every precalculated game :-)

**Comment by TobiasBielefeld** [comment]
yes the "developer automove" was meant to test the winnable games. Thanks for the hint with Pyramid, I found out that there is a timing issue related to the "auto move kings" option.

And the other points will be added as you suggested, and thanks for the nice words :)
